sql/_13_issues/_24_1h/cases/cbrd_25361.sql

```diff
@@ -0,0 +1,233 @@
+(SELECT * FROM cte_a ORDER BY i)
+DIFFERENCE
+(SELECT * FROM cte_a ORDER BY i);
+
+show trace;
+
+-- 1.4 Single CTE: DIFFERENCE ALL
+WITH cte_a AS (SELECT * FROM tbl_a ORDER BY 1 LIMIT 20)
+(SELECT * FROM cte_a ORDER BY i)
+DIFFERENCE ALL
+(SELECT * FROM cte_a ORDER BY i);
+
+show trace;
+
+-- 1.5 Single CTE: INTERSECTION
+WITH cte_a AS (SELECT * FROM tbl_a ORDER BY 1 LIMIT 20)
+(SELECT * FROM cte_a ORDER BY i)
+INTERSECTION
+(SELECT * FROM cte_a ORDER BY i);
+
+show trace;
+
+-- 1.6 Single CTE: INTERSECTION ALL
+WITH cte_a AS (SELECT * FROM tbl_a ORDER BY 1 LIMIT 20)
+(SELECT * FROM cte_a ORDER BY i)
+INTERSECTION ALL
+(SELECT * FROM cte_a ORDER BY i);
+
+show trace;
+
+-- 1.7 Multiple CTEs: UNION ALL with Uncorrelated Subqueries
+drop table if exists tbl_b;
+
+create table tbl_b ( a int, b varchar, c varchar);
+
+insert into tbl_b select rownum, rownum, rownum 
+from db_class a, db_class c, db_class d limit 100000;
+
+set trace on;
+
+WITH cte_a AS (SELECT * FROM tbl_a ORDER BY 1 LIMIT 20),
+     cte_b AS (SELECT * FROM tbl_b ORDER BY 1 LIMIT 20)
+(SELECT * FROM cte_a WHERE i < 50 ORDER BY i)
+UNION ALL
+(SELECT * FROM cte_b WHERE a > 50 ORDER BY a);
+
+show trace;
+drop table tbl_b;
+
+-- 1.8 Multiple CTEs: UNION with Uncorrelated Subqueries
+set trace on;
+
+WITH cte_a AS (SELECT * FROM tbl_a WHERE i < 50 ORDER BY 1 LIMIT 20),
+     cte_b AS (SELECT * FROM tbl_a WHERE i >= 50 ORDER BY 1 LIMIT 20)
+(SELECT * FROM cte_a WHERE j < 30 ORDER BY i)
+UNION
+(SELECT * FROM cte_b WHERE k > 50 ORDER BY i);
+
+show trace;
+
+-- 1.9 Multiple CTEs: DIFFERENCE with Uncorrelated Subqueries
+set trace on;
+
+WITH cte_a AS (SELECT * FROM tbl_a WHERE i < 50 ORDER BY 1 LIMIT 20),
+     cte_b AS (SELECT * FROM tbl_a WHERE i >= 50 ORDER BY 1 LIMIT 20)
+(SELECT * FROM cte_a WHERE i < 20 ORDER BY i)
+DIFFERENCE
+(SELECT * FROM cte_b WHERE k > 30 ORDER BY i);
+
+show trace;
+
+-- 1.10 Multiple CTEs: DIFFERENCE ALL with Uncorrelated Subqueries
+set trace on;
+
+WITH cte_a AS (SELECT * FROM tbl_a WHERE i < 50 ORDER BY 1 LIMIT 20),
+     cte_b AS (SELECT * FROM tbl_a WHERE i >= 50 ORDER BY 1 LIMIT 20)
+(SELECT * FROM cte_a WHERE i < 20 ORDER BY i)
+DIFFERENCE ALL
+(SELECT * FROM cte_b WHERE k > 30 ORDER BY i);
+
+show trace;
+
+-- 1.11 Multiple CTEs: INTERSECTION with Uncorrelated Subqueries
+set trace on;
+
+WITH cte_a AS (SELECT * FROM tbl_a WHERE i < 50 ORDER BY 1 LIMIT 20),
+     cte_b AS (SELECT * FROM tbl_a WHERE i >= 50 ORDER BY 1 LIMIT 20)
+(SELECT * FROM cte_a WHERE j < 30 ORDER BY i)
+INTERSECTION
+(SELECT * FROM cte_b WHERE k > 50 ORDER BY i);
+
+show trace;
+
+-- 1.12 Multiple CTEs: INTERSECTION ALL with Uncorrelated Subqueries
+set trace on;
+
+WITH cte_a AS (SELECT * FROM tbl_a WHERE i < 50 ORDER BY 1 LIMIT 20),
+     cte_b AS (SELECT * FROM tbl_a WHERE i >= 50 ORDER BY 1 LIMIT 20)
+(SELECT * FROM cte_a WHERE j < 30 ORDER BY i)
+INTERSECTION ALL
+(SELECT * FROM cte_b WHERE k > 50 ORDER BY i);
+
+show trace;
+
```

Comment:
--1.13 Use limit
WITH cte1 AS (SELECT i FROM t1 limit 1, 10)
(SELECT * FROM cte1)
UNION ALL
(SELECT * FROM cte1)
UNION
(SELECT * FROM cte1)
DIFFERENCE
(SELECT * FROM cte1)
INTERSECTION
(SELECT * FROM cte1)
;

--1.14 Use limit offset
WITH cte1 AS (SELECT i FROM t1 limit 3 offset 7)
(SELECT * FROM cte1)
UNION ALL
(SELECT * FROM cte1)
UNION
(SELECT * FROM cte1)
DIFFERENCE
(SELECT * FROM cte1)
INTERSECTION
(SELECT * FROM cte1)
;

--1.15 Use rownum
WITH cte1 AS (SELECT i FROM t1 where rownum < 5 )
(SELECT * FROM cte1)
UNION ALL
(SELECT * FROM cte1)
UNION
(SELECT * FROM cte1)
DIFFERENCE
(SELECT * FROM cte1)
INTERSECTION
(SELECT * FROM cte1)
;

--1.16 Use orderby_num()
WITH cte1 AS (SELECT i FROM t1 order by 1 for orderby_num() between 3 and 5)
(SELECT * FROM cte1)
UNION ALL
(SELECT * FROM cte1)
UNION
(SELECT * FROM cte1)
DIFFERENCE
(SELECT * FROM cte1)
INTERSECTION
(SELECT * FROM cte1)
;